### ISSUE TITLE
Expose deriver for use by other derivers?

### ISSUE THREAD

**Issue by apatil** [ocaml-ppx/ppx_deriving_yojson#65]
I'm working on a library, https://github.com/apatil/aplomb, that involves a ppx deriver. The deriver needs to emit `to_yojson` as well as a few other functions that are specific to my library. 

Put another way, in the example in the https://github.com/apatil/aplomb/README.md, I currently have `[@@deriving yojson,aplomb]`. I'd like to shorten that to `[@@deriving aplomb]` but still have ppx_deriving_yojson generate `to_yojson`.

Would it make sense to expose ppx_deriving_yojson's deriver in the library so that derivers like mine can use and augment it, or is there an easier way to do what I want? 

@whitequark I'd be happy to submit a patch if it makes sense, I see from other issues that you're crunched for time.

**Comment by whitequark** [comment]
I agree this is valuable. Please submit a patch. I'll review the overall architecture.
